### ISSUE TITLE
Show keyboard shortcuts in menus and tooltips

### ISSUE THREAD

**Issue by jasmussen** [WordPress/gutenberg#2980]
Aside from improving keyboard shortcuts (see #71, #2960) and others, let's discuss how we can surface these keyboard shortcuts to people. In this ticket I'd like to propose three ways we can do this:

## 1. Show keyboard shortcuts in menus

<img width="575" alt="screen shot 2017-10-11 at 10 10 15" src="https://user-images.githubusercontent.com/1204802/31429055-36770944-ae6d-11e7-842f-c7ef4b5abc03.png">

This is a mockup of a redesigned side menu, and the shortcuts are just bad suggestions from me, but the idea is that any menu that holds items with keyboard shortcuts show these shortcuts on the right.

## 2. Show keyboard shortcuts in tooltips

<img width="156" alt="screen shot 2017-10-11 at 10 11 00" src="https://user-images.githubusercontent.com/1204802/31429137-76098bf4-ae6d-11e7-8e33-2c954afb130a.png">

This is helpful in the same way tooltips are helpful.

## 3. Show a keyboard shortctus sheet when you press Shift+?

This is a common pattern in Google properties, press Shift+? to bring up a sheet that lists all keyboard shortcuts. Here's how it looks in Inbox:

<img width="1011" alt="screen shot 2017-10-11 at 10 19 14" src="https://user-images.githubusercontent.com/1204802/31429216-bd859540-ae6d-11e7-9391-14143ef573bf.png">

Thoughts?

**Comment by karmatosed** [comment]
I really like this, it's a strong usability improvement for a wide range of users. I also really like the contextual information of having them 'there' when you need them.

**Comment by vladanost** [comment]
@jasmussen I hope this lands soon, it is a great feature.
Is there a comprehensive list of Gutenberg shortcuts available anywhere at the moment?
Maybe we could have a separate issue for keyboard shortcut requests.
Here is one: Shortcut for switching from Text to Visual editor mode, really handy for development.

**Comment by jasmussen** [comment]
There isn't a keyboard shortcuts list yet, though there's a "todo list" at #71. We should use that ticket to both check off which are done, and which are still missing, I suppose.

**Comment by afercia** [comment]
@jasmussen when you have a chance: going through this issue during today's accessibility bug-scrub: and wondering what's left here?

**Comment by jasmussen** [comment]
I have sort of on my "secret agenda" to create a native extension that uses the incoming Modal API to create a shortcut sheet like the one mocked up in item 3. But even if I never got to that, it's fine to ship without it. 

We're missing some shortcuts + indicators in side menus, though I suppose this might be best handled in individual issues, like #7297 perhaps? If you feel that's the case, feel free to close this one.

**Comment by talldan** [comment]
@jasmussen Will close this now, as I think the three points have been addressed.
